devices: add lpc13xx family

lucab opened the pull request:
This adds `lpc13xx-pac` for the LPC13xx family of chips:
 - `lpc1311`
 - `lpc1313`
 - `lpc1342`
 - `lpc1343`

---

Vendor SVD contains duplicate names which do not work with `svd2rust`. These are the manual fixes that I had to make:
 - all `enumeratedValues`: dropped non-unique `ENUM` name labels
 - `EDGECT32B1_MAT1`: wrong name (`MAT0` duplication)
 - `EDGECT16B0_MAT1`: wrong name (`MAT0` duplication)
 - `BODINTVAL`: duplicate level names (`THE_INTERRUPT_ASSERT`)
 - `BODRSTLEV`: duplicate level names (`THE_INTERRUPT_ASSERT`)
 - `RILOC`: duplicate names (`SELECTS_RI_FUNCTION_` and `RESERVED_`)
 - `DCDLOC`: duplicate names (`RESERVED_`)
 - `SCKLOC`: duplicate names (`SELECTS_SCK0_FUNCTIO`)
 - `DSRLOC`: duplicate names (`SELECTS_DSR_FUNCTION_` and `RESERVED_`)

allexoll commented on lpc_part_table.yaml:
```diff
@@ -34,4 +34,15 @@ lpc81x-pac:
     members:
       - lpc810
       - lpc811
-      - lpc812
```
I think removing the lpc812 was a misclick there

allexoll commented:
Additionally, you should add a GA for this pac

allexoll commented:
for the remainder of the issues, I think you should try to fix them using svdtools/yamls as patches. you can find examples over at [stm32-rs](https://github.com/stm32-rs/stm32-rs/)

lucab committed:
devices: add lpc13xx family

This adds PACs for the lpc13xx family of chips:
 - lpc1311
 - lpc1313
 - lpc1342
 - lpc1343

lucab commented:
The remaining issue is a bug in a library used by `svdtools patch`: https://github.com/rust-embedded/svd/pull/216.

While the fix gets merged and percolate through dependencies, I'll opt-out XML linting for the moment.

lucab commented:
@allexoll when you are around, could you please re-trigger the CI here?

allexoll commented:
when svd dep has been published with your fix can you ping here so that we can re-enable the check on this family?
